wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```diff
@@ -102,20 +116,185 @@ class WalletTransactionMetadataProvider @Inject constructor(
         TODO("Not yet implemented")
     }
 
+    // should this be running in a separate thread each time, or the same as the caller?
```

Comment:
I have made a change following this:

> Since this one is run by the BlockchainService which can't handle suspend methods, we can expose a sync method in a runBlocking wrapper instead. Then the BlockchainService can run it in a dedicated handler or however else it wishes.